bundle/edu.gemini.pot/src/main/scala/edu/gemini/pot/sp/validator/SeqConstraint.scala

```diff
@@ -8,10 +8,15 @@ import scala.Some
 object SeqConstraint {
 
   // These types can occur in any sequence
-  val genericTypes = Types.initial
+  val genericTypes: Types = Types.initial
     .addBroad[ISPSeqComponent](OBSERVER)
     .addNarrow[ISPSeqComponent](ITERATOR_CALUNIT, ITERATOR_REPEAT)
 
+  def initialTypesForInstrument(ct: SPComponentType): Types = ct match {
+    case INSTRUMENT_GHOST => genericTypes - OBSERVER_DARK
+    case _                => genericTypes - OBSERVER_GHOST_DARK
+  }
```

Comment:
Ghost may be the only instrument like this in this codebase, but it would get awkward if we had Scorprio as well.  You'd need to subtract Scorpio from Ghost and Ghost from Scorpio (and both from everything else).

Another way to organize this perhaps would be to define a `Map[Instrument, Types]` of instrument-specific types.  Here you'd first define `standardObserveTypes: Types` with `OBSERVER_DARK` etc. All instruments would be assigned `standardObserveTypes` but you'd change the Ghost entry to have `OBSERVER_GHOST_DARK` etc. instead.

Then `genericTypes` could be defined as it is but (programmatically) subtracting everything in the instrument specific types map.  At that point `initialTypesForInstrument` would always be `genericTypes + instrumentTypes(ct)`.

Reply:
I agree that this could get sloppy quickly if we continue down this route. I'll make this change.

Reply:
This is the one part I'm slightly confused about. Do you mean:

1. In `SeqConstraint` have, say, a:
```
val instrumentSpecificTypes: Map[SPComponentType, Types]
```
where we have, say, `ITERATOR_GHOST` map to `standardObserveTypes - OBSERVER_DARK + OBSERVER_GHOST_DARK` etc? I guess I'm having difficulty differentiating between `standardObserveTypes` and `genericTypes`.

I feel like I'm missing something in the understanding here.

Reply:
I'm having trouble getting this to work, and I want to move on to other tasks, so I'm going to make a JIRA task for this and merge this and get back to it later, I think.